.github/workflows/release.yml

```diff
@@ -3,50 +3,256 @@ on:
+            target: x86_64-unknown-linux-gnu
+            build: pnpm build --target x86_64-unknown-linux-gnu --use-napi-cross
+    name: stable - ${{ matrix.settings.target }} - node@20
+    runs-on: ${{ matrix.settings.host }}
+    defaults:
+      run:
+        working-directory: packages/matchstick
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+      - name: setup pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9.5.0
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: pnpm
+
+      - name: Install
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+          targets: ${{ matrix.settings.target }}
+
+      - name: Cache cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ~/.napi-rs
+            .cargo-cache
+            target/
+          key: ${{ matrix.settings.target }}-cargo-${{ matrix.settings.host }}
+
+      - uses: goto-bus-stop/setup-zig@v2
+        if: ${{ contains(matrix.settings.target, 'musl') }}
         with:
-          fetch-depth: 0
-          token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Setup environment
-        uses: the-guild-org/shared-config/setup@main
+          version: 0.13.0
+
+      - name: Setup toolchain
+        run: ${{ matrix.settings.setup }}
+        if: ${{ matrix.settings.setup }}
+        shell: bash
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Setup node x86
+        uses: actions/setup-node@v4
+        if: matrix.settings.target == 'i686-pc-windows-msvc'
+        with:
+          node-version: 20
+          cache: pnpm
+          architecture: x86
+
+      - name: Install Protobuf on macOS
+        if: matrix.settings.host == 'macos-latest'
+        run: brew install protobuf
+
+      - name: Install Protobuf on Windows
+        if: matrix.settings.host == 'windows-latest'
+        run: choco install protoc
+      - name: Set PROTOC environment variable on Windows
+        if: matrix.settings.host == 'windows-latest'
+        run: echo 'PROTOC=C:\Program Files\Google Protobuf\bin\protoc.exe' >> $GITHUB_ENV
+
+      - name: Install OpenSSL and Protobuf
+        if: matrix.settings.host == 'ubuntu-latest'
+        run: sudo apt-get update && sudo apt-get install -y libssl-dev protobuf-compiler
+
+      - name: Cache custom OpenSSL
+        if: matrix.settings.host == 'ubuntu-latest'
+        uses: actions/cache@v4
+        id: openssl-cache
         with:
-          nodeVersion: 20
-          packageManager: pnpm
-          packageManagerVersion: 9.1.0
+          path: /usr/local/openssl-1.1.1o
+          key: openssl-1.1.1o
 
-      - name: Set variables
-        id: vars
+      - name: Install custom OpenSSL
```

Comment:
why do we need this?

Reply:
napi build process needs it, and for linux specifically there has been a common issue about the latest openssl version being broken, so this guy built a custom one that works:
https://www.reddit.com/r/archlinux/comments/ytj40j/opensslsys_building_issues_in_rust_for_spotifytui/

if you just do `apt-get openssl` you get this:
https://github.com/graphprotocol/graph-tooling/actions/runs/9945547835/job/27474045275